app/models/notification.rb

```diff
@@ -79,7 +87,47 @@ def unarchive_if_updated
   def update_from_api_response(api_response, unarchive: false)
     attrs = Notification.attributes_from_api_response(api_response)
     self.attributes = attrs
+    update_subject
     unarchive_if_updated if unarchive
     save(touch: false) if changed?
   end
+
+  private
+
+  def download_subject
+    user.github_client.get(subject_url)
+  end
+
+  def update_subject
+    return unless Octobox.config.fetch_subject
+    if subject
+      # skip syncing if the notification was updated around the same time as subject
+      return if updated_at - subject.updated_at < 2.seconds
+
+      case subject_type
+      when 'Issue', 'PullRequest'
+        remote_subject = download_subject
+        subject.state = remote_subject.merged_at.present? ? 'merged' : remote_subject.state
+        subject.save(touch: false) if subject.changed?
+      end
+    else
+      case subject_type
+      when 'Issue', 'PullRequest'
+        remote_subject = download_subject
+        create_subject({
```

Comment:
Realization: Should this code branch be `find_or_create_by`'s or similar? One notification from user A might create a subject and another notification from user B that *doesn't* have that subject associated with it should use the existing subject they share, not create their own (or is that what these meta methods do? I forget, I haven't used them in quite a while).